tests/rule_based_profiler/parameter_builder/test_regex_pattern_string_parameter_builder.py

```diff
@@ -0,0 +1,238 @@
+from typing import Set
+
+import pytest
+
+import great_expectations.exceptions.exceptions as ge_exceptions
+from great_expectations.data_context import DataContext
+from great_expectations.execution_engine.execution_engine import MetricDomainTypes
+from great_expectations.rule_based_profiler.parameter_builder import (
+    RegexPatternStringParameterBuilder,
+    regex_pattern_string_parameter_builder,
+)
+from great_expectations.rule_based_profiler.types import (
+    Domain,
+    ParameterContainer,
+    get_parameter_value_by_fully_qualified_parameter_name,
+)
+from tests.integration.profiling.rule_based_profilers.conftest import (
+    alice_columnar_table_single_batch,
+    alice_columnar_table_single_batch_context,
+    bobby_columnar_table_multi_batch_deterministic_data_context,
+)
+
+
+def test_regex_pattern_string_parameter_builder_instantiation():
+    candidate_regexes: Set[str] = {
+        r"^\d{1}$",
+        r"^\d{2}$",
+        r"^\S{8}-\S{4}-\S{4}-\S{4}-\S{12}$",
+    }
+
+    regex_pattern_string_parameter: RegexPatternStringParameterBuilder = (
+        RegexPatternStringParameterBuilder(
+            name="my_simple_regex_string_parameter_builder",
+        )
+    )
+
+    assert regex_pattern_string_parameter._threshold == 1.0
+    assert regex_pattern_string_parameter._candidate_strings == candidate_regexes
+    assert regex_pattern_string_parameter.CANDIDATE_STRINGS == candidate_regexes
+
```

Comment:
I'm wary of testing internals like this due to the possibility of the implementation changing. If the specific attributes change, the ending behavior can still be the same (but this test will fail). Ex. I rename `_candidate_strings` to `_candidate_strs`.

What is the behavior of this unit? What is it doing for the user and what behavior is that user relying on? If we can answer those questions, let's tailor our tests to meet those needs. If not, I don't think testing internals is the worst thing in the world. But one should do some with a certain degree of caution due to the point noted above.

Reply:
Hmm that's fair especially since this is the constructor. I'm just wary of relying too heavy on `assert self._private_attr` since it's caused a bit of pain with refactors elsewhere in the codebase.

Happy to leave this be based on your points @NathanFarmer

Reply:
We achieved 100% code coverage with these tests, and this line helps to that end, but yes it will require a change to this test if the defaults are changed. @cdkini

Reply:
In light of the discussion below, I'm leaning towards having `candidate_strings` be a required parameter for `RegexPatternStringParameterBuilder`... I think any default we would have to be really careful with, since it would be really easy to get 0% match if too specific, or 100% match if it is too broad (`".*"` or something similar).

How do you feel about changing the signature to have it be a required parameter?

Reply:
This is how I see it, but Alex is leaning towards including defaults based on his comments.